### PR TITLE
Add tests for parameter of 'None' for TPE

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation
Add test for #3190 

## Description of the changes
Add a variable sampled from categorical distribution of `[None]` in `test_get_observation_pairs()`

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@himkt Could you review this PR?

File: tests/samplers_tests/tpe_tests/test_sampler.py [hunk 1]
```diff
@@ -757,6 +757,8 @@ def test_get_observation_pairs() -> None:
     def objective(trial: Trial) -> float:
 
         x = trial.suggest_int("x", 5, 5)
+        z = trial.suggest_categorical("z", [None])
+        z  # Use z here
         if trial.number == 0:
             return x
```
Comment (HideakiImamura) [inline tests/samplers_tests/tpe_tests/test_sampler.py:763]: I think the first line is a bit confusing. For example, how about changing the following branch?
```suggestion
        if trial.number == 0:
            return x * int(z == None)
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have a minor comment. PTAL.

File: tests/samplers_tests/tpe_tests/test_sampler.py (hunk 1 above)
Comment (contramundum53) [reply tests/samplers_tests/tpe_tests/test_sampler.py]: Thanks, I applied this change.

**Review by HideakiImamura — APPROVED** [review]
Thanks, LGTM!

**Review by himkt — APPROVED** [review]

**Comment by himkt** [comment]
Thank you @contramundum53, LGTM!
